src/ResourceManager/Compute/Commands.Compute/Models/AzureDiskEncryptionStatusContext.cs

```diff
@@ -3,16 +3,24 @@
 
 namespace Microsoft.Azure.Commands.Compute.Models
 {
+    enum EncryptionStatus
+    {
+        Encrypted,
+        NotEncrypted,
+        Unknown
+    }
+
     class AzureDiskEncryptionStatusContext
     {
-        public bool OsVolumeEncrypted { get; set; }
+        public EncryptionStatus OsVolumeEncrypted { get; set; }
         public DiskEncryptionSettings OsVolumeEncryptionSettings { get; set; }
-
-        [JsonIgnore]
-        public string OsVolumeEncryptionSettingsText
-        {
-            get { return JsonConvert.SerializeObject(OsVolumeEncryptionSettings, Formatting.Indented); }
-        }
-        public bool DataVolumesEncrypted { get; set; }
+        public EncryptionStatus DataVolumesEncrypted { get; set; }
+    }
+    class AzureDiskEncryptionStatusLinuxContext
+    {
```

Comment:
Does it mean, the service returns both OsVolumeEncryptionSettings and DataVolumeEncryptionSettings for Windows in response body, but Powershell hides DataVolumeEncryptionSettings part?  I don't think it is good to hide any information.  If it is required to be hidden, the service should be fixed.